hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -170,6 +170,9 @@ extends Importer:
           N
       case N =>
         N
+    case S(_: TopLevelSymbol) => ctx.get(nme.name).flatMap(_.symbol) match
+      case S(sym: FieldSymbol) => S(sym)
```

Comment:
I think this logic is incorrect. A top-level member may very well be shadowed by a local one in `ctx`.

You may want to have a read of https://github.com/hkust-taco/mlscript/pull/242#discussion_r1861679152